phac-nml/irida-next
# Switch from Azure AD to Entra ID

## What does this PR do and why?
_Describe in detail what your merge request does and why.
Fixes #840

Azure AD is being deprecated in favour of Entra ID. They function identically but with a different name.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Clean your db `./scripts/clean_dev.sh`
2. edit `config/authentication/auth_config.yml` to add `entra_id` to  `development:` -> `omniauth_providers:`
3. edit your credentials file

`EDITOR="vim --nofork" bin/rails credentials:edit --environment development`

Add the following fields. Message me directly for the secret credentials.

```
entra_id:
  client_id: secret
  client_secret: secret
  tenant_id: secret
```

4. run irida next
5. click the sign in button for Entra ID
6. Go to http://localhost:3000/-/profile  and verify that a user account has been created for you with the relevant provider info. 
![image](https://github.com/user-attachments/assets/7e812168-26bf-45f9-868b-9769f49ea31d)

## Testing the Migration
1. on the `main` branch, clean your db with `./scripts/clean_dev.sh`
2. edit `auth_config.yml` to have `omniauth_providers: [azure_activedirectory_v2]` for `development:`
3. edit your credentials for development to have the required fields under `azure:`
4. run irida next, and sign in with Azure AD
5. verify you have the `azure_activedirectory_v2` provider on your profile
6. Note your Authentication Provider UID
7. sign out of your account
8. stop irida next, checkout the `oauth_entra_id` branch
9. run the migration with `rails info db:migrate`
10. run irida next
11. sign in using Entra ID
12. verify that your profile reflects the change to entra_id
13. verify that your Authentication Provider UID is now the concatenation of the tenant_id and your old Authentication Provider UID

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
